modules/concurrency/include/hephaestus/concurrency/spinner_state_machine.h

```diff
@@ -0,0 +1,30 @@
+//=================================================================================================
+// Copyright (C) 2023-2024 HEPHAESTUS Contributors
+//=================================================================================================
+
+#pragma once
```

Comment:
let's keep this in the same file, it is much easier to read without having to jump between files together they are less than 100 loc